rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,101 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required information to interact with both Teleport access and the Opsgenie API.
+
+```
+[opsgenie]
+api_key = "path/to/key" # File containing Opsgenie API Key
+opsgenie_addr = "example.app.opsgenie.com" # Address of Opsgenie
+severity = "2" # Severity to create Opsgenie alerts with
+```
+
+### Getting an Opsgenie API key
+
+In the Opsgenie web UI go to Settings -> App settings -> API key management. Create key with Read, Create and Update access.
+
+### Executing
+
+The plugin will be started using a command of the form
+
+```
+Teleport opsgenie start –config <config file location>
+```
+
+## UX
+
+Once an access request has been created, the Opsgenie plugin will create an alert in the service specified in the request annotation using the Opsgenie Alert API Create endpoint. 
+
+The appropriate on call responder can then click into the provided link and approve or deny the access request.
```

Comment:
Will this link to the web UI for the access requests? In the past we've had security issues with links that directly approved or denied access requests in plugins.

Reply:
Somewhat related.

The JIRA plugin updates the access request based on jira events (webhooks) when the user moves the ticket (using the JIRA Web App) to Approve or Deny column.

I'm not sure if this is possible with Opsgenie, but we should be clear about this: Opsgenie only reads Teleport state, and never writes to it.

There's only one thing that could trigger an approval: auto approval (which, IMO, should be configurable).
However, that isn't triggered by Opsgenie itself but from the plugin process

Reply:
That was the intention. Do you have any more info on the past security issues doing this and how they were avoided?

Reply:
I think what @espadolini is referring to, is the link should take the user to the page for this access request which has approve/deny buttons the user should click explicitly. Rather than having a link that just auto-approves when you click on it (which is prone to phishing attacks). I don't even think we have ability to generate such links but might be worth calling this out explicitly in the RFD.